# Assignment 1 
## Hypothesis Testing


### Question 1
In the past, the mean running time for a certain type of flashlight battery has been
8.5 hours. The manufacturer has introduced a change in the production method which
he hopes has increased the mean running time. The mean running time for a random
sample of 40 light bulbs was 8.7 hours. Do the data provide sufficient evidence to
conclude that the mean running time of all light bulbs, μ, has increased from the
previous mean of 8.5 hours? Assume that σ = 0.5 hours.

In [57]:
import numpy as np
import pandas as pd
import math

In [55]:
# Given data

# Sample mean running time
sample_mean = 8.7  

# Population mean running time
population_mean = 8.5

 # Population standard deviation
population_stddev = 0.5 

# Sample size
sample_size = 40 

# Significance level
alpha = 0.05  

In [59]:
# Calculate the z-score
z_score = (sample_mean - population_mean) / (population_stddev / math.sqrt(sample_size))

In [60]:
z_score

2.529822128134694

In [62]:
# Assuming a one-tailed test (since we're testing if the mean has increased)
p_value = 1 - 0.5 * (1 + math.erf(z_score / math.sqrt(2)))

In [63]:
p_value

0.0057060181930009835

In [64]:
# Print the results
print("Z-score:", z_score)
print("P-value:", p_value)

Z-score: 2.529822128134694
P-value: 0.0057060181930009835


In [65]:
# Compare p-value with significance level
if p_value < alpha:
    print("Reject the null hypothesis. There is sufficient evidence to conclude that the mean running time has increased.")
else:
    print("Fail to reject the null hypothesis. There is not sufficient evidence to conclude that the mean running time has increased.")

Reject the null hypothesis. There is sufficient evidence to conclude that the mean running time has increased.


Based on the conducted one-sample z-test with a significance level of 0.05, the obtained p-value of 0.0057060181930009835 indicates that we reject the null hypothesis. Therefore, there is sufficient evidence to conclude that the mean running time of all flashlight batteries has increased from the previous mean of 8.5 hours.

## Question 2
The forced vital capacity (FVC) is often used by physicians to assess a person’s ability
to move air in and out of their lungs. For adult males, the average FVC is 5.0 liters. A
researcher wants to perform a hypothesis test to determine whether the average FVC
for women differs from this value. The mean forced vital capacity for a random sample
of 85 women was 4.8 liters. Do the data provide sufficient evidence to conclude that
the mean FVC for women differs from the mean value for men of 5.0 liters? Assume
that σ = 0.9 liters.

In [66]:
import numpy as np
import pandas as pd
import math

In [67]:
# Given data
sample_mean = 4.8  # Sample mean FVC for women
population_mean = 5.0  # Population mean FVC for men
population_stddev = 0.9  # Population standard deviation
sample_size = 85  # Sample size
alpha = 0.05  # Significance level

In [68]:
# Calculate the z-score
z_score = (sample_mean - population_mean) / (population_stddev / math.sqrt(sample_size))

In [69]:
z_score

-2.0487876571761987

In [70]:
# Find the two-tailed p-value using z-table (standard normal distribution table)
# Since it's a two-tailed test, we need to consider both tails
p_value = 2 * (1 - 0.5 * (1 + math.erf(abs(z_score) / math.sqrt(2))))

In [71]:
p_value

0.04048288322366167

In [72]:
# Print the results
print("Z-score:", z_score)
print("P-value:", p_value)

Z-score: -2.0487876571761987
P-value: 0.04048288322366167


In [73]:
# Compare p-value with significance level
if p_value < alpha:
    print("Reject the null hypothesis. There is sufficient evidence to conclude that the mean FVC for women differs from the mean value for men.")
else:
    print("Fail to reject the null hypothesis. There is not sufficient evidence to conclude that the mean FVC for women differs from the mean value for men.")

Reject the null hypothesis. There is sufficient evidence to conclude that the mean FVC for women differs from the mean value for men.


Based on the conducted one-sample z-test with a significance level of 0.05, the obtained p-value of 0.04048288322366167 indicates that we reject the null hypotesis. Therefore, there is sufficient evidence to conclude that the mean forced vital capacity (FVC) for women differs from the mean value for men of 5.0 liters.









## Question 3
A manufacturer claims that the mean amount of juice in its 16 ounce bottles is 16.1
ounces. A consumer advocacy group wants to perform a hypothesis test to determine
whether the mean amount is actually less than this. The mean volume of juice for a
random sample of 70 bottles was 15.94 ounces. Do the data provide sufficient evidence
to conclude that the mean amount of juice for all 16-ounce bottles, μ, is less than 16.1
ounces? Assume that σ = 0.9 ounces.

In [77]:
import math
from scipy.stats import norm

Since we are interested in whether the mean amount of juice is less than 16.1 ounces, we use the CDF to find the probability in the left tail of the distribution.

In [78]:
# Given data
sample_mean = 15.94  # Sample mean amount of juice
population_mean = 16.1  # Population mean amount of juice
population_stddev = 0.9  # Population standard deviation
sample_size = 70  # Sample size
alpha = 0.05  # Significance level

In [79]:
# Calculate the z-score
z_score = (sample_mean - population_mean) / (population_stddev / math.sqrt(sample_size))

In [80]:
z_score

-1.4873956027272632

In [81]:
# Calculate the one-tailed p-value using the cumulative distribution function (CDF)
p_value = norm.cdf(z_score)

In [82]:
p_value

0.06845517880167845

In [83]:
# Print the results
print("Z-score:", z_score)
print("P-value:", p_value)

Z-score: -1.4873956027272632
P-value: 0.06845517880167845


In [84]:
# Compare p-value with significance level
if p_value < alpha:
    print("Reject the null hypothesis. There is sufficient evidence to conclude that the mean amount of juice for all 16-ounce bottles is less than 16.1 ounces.")
else:
    print("Fail to reject the null hypothesis. There is not sufficient evidence to conclude that the mean amount of juice for all 16-ounce bottles is less than 16.1 ounces.")

Fail to reject the null hypothesis. There is not sufficient evidence to conclude that the mean amount of juice for all 16-ounce bottles is less than 16.1 ounces.


The p-value of 0.06845517880167845 suggests that the mean amount of juice for 16-ounce bottles is not less than 16.1 ounces at a significance level of 0.05